frontend/public/js/util.js

```diff
@@ -10,4 +10,31 @@
       function () { return $(this).data(prop) == val; }
     );
   };
-})(window.jQuery);
\ No newline at end of file
+})(window.jQuery);
+
+
+JSON.flatten = function(data) {
+    var result = {};
+    function recurse (cur, prop) {
+        if (Object(cur) !== cur) {
+            result[prop] = cur;
+        } else if (Array.isArray(cur)) {
+          /*
+             for(var i=0, l=cur.length; i<l; i++)
```

Comment:
do you plan to use the commented out code? if not, let's remove it!

re: flattening: currently filtering works for top-level keys right. are you planning to add in nested filtering? i think it would make sense because a lot of the metrics are nested